### PR TITLE
Check type compability when dragging and dropping parameters into filter/post filter

### LINKED ISSUE #1335: Bug: Check type compability when dragging and dropping parameters into filter/post filter
### Similar issues

- [X] I have searched and found no existing similar issues

### How are you using Studio?

Legend Query

### Current and expected behavior

when dragging and dropping a parameter into the filter/post filter condition value, we should do type validation.

### Steps to reproduce

_No response_

### Model data

_No response_

### Environment

```text
chrome
```

### Possible solution and workaround

_No response_

### Contribution

- [X] I would like to work on the fix for this issue

### PR DESCRIPTION
**Pull request by YannanGao-gs**
## Summary
Fixes #1335 

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

https://user-images.githubusercontent.com/73408381/180880199-fb05c18e-65d0-4d05-bcdb-1cc3e2e9008e.mov

### CODE REVIEW

**Comment by gayathrir11** [comment]
Is handling check by multiplicity in the scope of this issue?

**Comment by YannanGao-gs** [comment]
> Is handling check by multiplicity in the scope of this issue?

Post filter/filter will switch to use `exist` if LHS's multiplicity is MANY, which means that we only need to check if parameter's multiplicity is 0 or 1 or [0..1] if we need to take care of multiplicity.

@akphi @MauricioUyaguari  May I know ur opinions too?

Thanks!

File: .changeset/three-pants-live.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-query': patch
+---
+
+Check type compabilty when dragging and dropping parameters into filter/post filter ([#1335](https://github.com/finos/legend-studio/issues/1335)).
```
Comment (akphi) [inline .changeset/three-pants-live.md:5]:
```suggestion
Check type compatibility when dragging and dropping parameters into `filter/post-filter` panel ([#1335](https://github.com/finos/legend-studio/issues/1335)).
```

File: packages/legend-query/src/components/QueryBuilderFilterPanel.tsx [hunk 2]
```diff
@@ -188,9 +193,42 @@ const QueryBuilderFilterConditionEditor = observer(
     // Drag and Drop on filter condition value
     const handleDrop = useCallback(
       (item: QueryBuilderParameterDragSource): void => {
-        node.condition.setValue(item.variable.parameter);
+        const NUMERIC_PRIMITIVE_TYPES = [
```
Comment (akphi) [inline packages/legend-query/src/components/QueryBuilderFilterPanel.tsx:196]: Seems like we repeat this piece of logic a lot in the operators. Let's try to see if we could abstract this logic out. Thanks!

**Review by akphi — CHANGES_REQUESTED** [review]
You need to rebase

File: packages/legend-application-query/src/components/QueryBuilderFilterPanel.tsx [hunk 3]
```diff
@@ -196,9 +200,21 @@ const QueryBuilderFilterConditionEditor = observer(
...
+          isTypeCompatibleWithConditionValueType(
+            item.variable.parameter.genericType?.value.rawType,
+            node.condition.propertyExpressionState.propertyExpression.func
+              .genericType.value.rawType,
+          )
+        ) {
+          node.condition.setValue(item.variable.parameter);
+        } else {
+          applicationStore.notifyWarning(
+            'Parameter type and filter condition value type mismatches.',
```
Comment (MauricioUyaguari) [inline packages/legend-application-query/src/components/QueryBuilderFilterPanel.tsx:213]:
````suggestion

`Incompatible parameter type ${type}. ${type} is not compatible with type ${node.condition.value.type}`

````

File: packages/legend-application-query/src/components/QueryBuilderPostFilterPanel.tsx [hunk 4]
```diff
@@ -421,9 +425,20 @@ const QueryBuilderPostFilterConditionEditor = observer(
     // Drag and Drop on filter condition value
     const handleDrop = useCallback(
       (item: QueryBuilderParameterDragSource): void => {
-        node.condition.setValue(item.variable.parameter);
+        if (
+          isTypeCompatibleWithConditionValueType(
+            item.variable.parameter.genericType?.value.rawType,
+            guaranteeNonNullable(node.condition.columnState.getReturnType()),
```
Comment (MauricioUyaguari) [inline packages/legend-application-query/src/components/QueryBuilderPostFilterPanel.tsx:431]: What does this mean for derived projeciton columns. 
We should probably escape the check for those if not known. I wouldn't do a `guaranteeNonNullable`

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]

File: packages/legend-application-query/src/components/QueryBuilderPostFilterPanel.tsx (hunk 4 above)
Comment (YannanGao-gs) [reply packages/legend-application-query/src/components/QueryBuilderPostFilterPanel.tsx]: getReturnType is an abstract function that will be overrode by `QueryBuilderDerivationProjectionColumnState` and `QueryBuilderSimpleProjectionColumnState`.

**Review by MauricioUyaguari — APPROVED** [review]

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**